jflex/src/test/java/jflex/state/StateSetQuickcheck.java

```diff
@@ -157,14 +159,26 @@ public void removeRemoves(
   @Property
   public void removeAdd(
       @Size(max = 90) @InRange(minInt = 0, maxInt = 100) StateSet s,
-      @InRange(minInt = 0, maxInt = 100) int e) {
+      @InRange(minInt = 0, maxInt = 100) int e,
+      @From(OffsetGen.class) int largeOffset) {
     assumeTrue(s.hasElement(e));
     StateSet sPre = new StateSet(s);
     s.remove(e);
     assertThat(sPre.contains(s)).isTrue();
     assertThat(s).isNotEqualTo(sPre);
     s.addState(e);
     assertThat(s).isEqualTo(sPre);
+
+    // add larger state value to force resize
+    if ((Integer.MAX_VALUE - largeOffset) <= s.bits.length) { // avoid overrun wrap
+      largeOffset = Integer.MAX_VALUE - s.bits.length;
+    }
+    int largerState =
+        ((s.bits.length + largeOffset) << StateSet.BITS) & Integer.MAX_VALUE; // & resets sign bit
+    s.addState(largerState);
+    assertThat(s).contains(largerState);
+    s.remove(largerState);
+    assertThat(s).isEqualTo(sPre);
```

Comment:
Adding something that forces resize is nice, but I'm not sure we want a resize in every single test case. I think I'd prefer to do this one in a separate property.

Also, the computation here is very dependent on the implementation of `StateSet` (i.e. a bit too white box). It'd probably be nicer to add something to the interface of StateSet that tells us what state would force a resize or what max state can currently be represented, and then we can pick the test case based on that interface.